runhouse/resources/hardware/cluster.py

```diff
@@ -1645,3 +1658,40 @@ def _check_for_child_configs(cls, config: dict):
             config["default_env"] = default_env
 
         return config
+
+    def _is_den_authed(self, change_type: str):
```

Comment:
for simplicity maybe we remove this separate method and just check `if self.den_auth` in-line? can still add logs if appropriate